ArduPlane/GCS_Mavlink.cpp

```diff
@@ -791,13 +792,197 @@ MAV_RESULT GCS_MAVLINK_Plane::handle_command_int_do_reposition(const mavlink_com
+        return MAV_RESULT_ACCEPTED;
+    }
+
+     case MAV_CMD_GUIDED_CHANGE_HEADING: {
+
+        // command is only valid in guided
+        if (plane.control_mode != &plane.mode_guided) {
+            gcs().send_text(MAV_SEVERITY_WARNING,"REJECTED - not in guided mode");
+            return MAV_RESULT_TEMPORARILY_REJECTED;
+        }
+
+         // don't accept packets out a [0-360) degree range
+        if (packet.param2 < 0.0f || packet.param2 >= 360.0f) {
+            gcs().send_text(MAV_SEVERITY_WARNING,"REJECTED - packet.param2 out of range 0-359.9");
+            return MAV_RESULT_TEMPORARILY_REJECTED;
+        }
+
+         float new_target_heading = radians(wrap_180(packet.param2));
+
+         // reject duplicate heading requests if our target-heading has been set to anything except it's default value of -1
+        if (( plane.guided_state.target_heading > -0.001 ) && (fabs(new_target_heading - plane.guided_state.target_heading)<0.001)) { // compare two floats as near-enough
+            gcs().send_text(MAV_SEVERITY_WARNING,"REJECTED - dup heading request");
+            return MAV_RESULT_TEMPORARILY_REJECTED;
+        }
+
+         if ( int(packet.param1) == HEADING_TYPE_COURSE_OVER_GROUND) { // compare as nearest int
+            plane.guided_state.target_heading_type = GUIDED_HEADING_COG;
+            plane.prev_WP_loc = plane.current_loc;
+            gcs().send_text(MAV_SEVERITY_WARNING,"COURSE_OVER_GROUND");
+        }
+        else if (int(packet.param1) == HEADING_TYPE_HEADING) { // compare as nearest int
+            plane.guided_state.target_heading_type = GUIDED_HEADING_HEADING;
+            gcs().send_text(MAV_SEVERITY_WARNING,"HEADING");
+        } else {
+            // unknown heading track type
+            gcs().send_text(MAV_SEVERITY_WARNING,"REJECTED");
+            return MAV_RESULT_TEMPORARILY_REJECTED;
+        }
+
+         plane.g2.guidedHeading.reset_I();
+
+         plane.guided_state.target_heading = new_target_heading;
+        plane.guided_state.target_heading_accel_limit = MAX(packet.param3, 0.05f);
+        plane.guided_state.target_heading_time_ms = AP_HAL::millis();
+        ::printf("ACCEPTED hdg:%f -> %f\n",packet.param2,packet.param3);
+        gcs().send_text(MAV_SEVERITY_WARNING,"ACCEPTED hdg");
+        return MAV_RESULT_ACCEPTED;
+    }
+#endif // OFFBOARD_GUIDED == ENABLED
+
+
+  }
+  // anything else ...
+  return MAV_RESULT_UNSUPPORTED;
+
+}
+
 MAV_RESULT GCS_MAVLINK_Plane::handle_command_int_packet(const mavlink_command_int_t &packet)
 {
+
+    plane.Log_Write_MavCmdI(packet);
+
     switch(packet.command) {
 
     case MAV_CMD_DO_REPOSITION:
         return handle_command_int_do_reposition(packet);
 
+    // special 'slew-enabled' guided commands here... for speed,alt, and direction commands
+    case MAV_CMD_GUIDED_CHANGE_SPEED:
+        return handle_command_int_guided_slew_commands(packet);
+    case MAV_CMD_GUIDED_CHANGE_ALTITUDE:
```

Comment:
Just combine the 3 cases so they fallthrough

Reply:
will fix on next attempt.